packages/core/core-flows/src/draft-order/steps/create-draft-order-shipping-method-adjustments.ts

```diff
@@ -0,0 +1,54 @@
+import { Modules } from "@medusajs/framework/utils"
+import { createStep, StepResponse } from "@medusajs/framework/workflows-sdk"
+import {
+  CreateShippingMethodAdjustmentDTO,
+  IOrderModuleService,
+} from "@medusajs/types"
+
+export const createDraftOrderShippingMethodAdjustmentsStepId =
+  "create-draft-order-shipping-method-adjustments"
+
+interface CreateDraftOrderShippingMethodAdjustmentsStepInput {
+  shippingMethodAdjustmentsToCreate: CreateShippingMethodAdjustmentDTO[]
+}
+
+export const createDraftOrderShippingMethodAdjustmentsStep = createStep(
+  createDraftOrderShippingMethodAdjustmentsStepId,
+  async function (
+    data: CreateDraftOrderShippingMethodAdjustmentsStepInput,
+    { container }
+  ) {
+    const { shippingMethodAdjustmentsToCreate = [] } = data
+
+    if (!shippingMethodAdjustmentsToCreate?.length) {
+      return new StepResponse(void 0, [])
+    }
+
+    const service = container.resolve<IOrderModuleService>(Modules.ORDER)
+
+    const shippingMethodAdjustments =
+      await service.createOrderShippingMethodAdjustments(
+        shippingMethodAdjustmentsToCreate
+      )
```

Comment:
question: Should we also filter out `undefined` amounts here, like we do for line items/